File: 70-climbing-stairs/stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 0: return 0
        if n <= 2: return n

        one_step = 2
        two_steps = 1
        count = 0

        for i in range(2,n):
            count = one_step + two_steps
            two_steps = one_step
            one_step = count
        
        return count
    

File: 70-climbing-stairs/stairs.ts
function climbStairs(n: number): number {
    if(n <= 0) return 0;
    if(n <= 2) return n;
    
    let one_step = 2;
    let two_steps = 1;
    let count = 0;
    
    for(let i=2; i<n; i++){
    	count = one_step + two_steps;
    	two_steps = one_step;
        one_step = count;
    }
    return count;
};